lib/ConvoyLeader/src/App.cpp

```diff
@@ -244,8 +262,44 @@ void App::loop()
     /* Process MQTT Communication */
     m_mqttClient.process();
 
-    /* Set new target speeds. */
-    sendSpeedSetpoints();
+    /* Process Platoon Controller */
+    m_platoonController.process();
+}
+
+void App::currentVehicleChannelCallback(const VehicleData& vehicleData)
+{
+    Waypoint vehicleDataAsWaypoint;

```

Comment:
As Waypoint and VehicleData are defined identically, could the following be used to prevent copying?
`Waypoint vehicleDataAsWaypoint= reinterpret_cast<const Waypoint&>(vehicleData);`

Reply:
Not good practice. If one of them changes, the program will go crazy.

To achieve this, you need to e.g.
```cpp
typedef Waypoint VehicleData;
```

And in this case, a static_cast is enough.

Reply:
I would like to implement your solution, however Waypoint is required in the PlatoonService layer, and VehicleData is shared with RadonUlzer at an Application level, so I think the only way in this case is manual copy